docs/sources/shared/troubleshoot-query.md

```diff
@@ -242,3 +242,326 @@ The `count_over_time` function doesn't use unwrapped values - it just counts log
+
+* **Add more specific filters** to return fewer results:
+
+   ```logql
+   {app="foo"} |= "error" 
+   ```
+
+* **Reduce the time range** of the query.
+
+* **Increase the limit** if needed:
+
+   ```yaml
+   limits_config:
+     max_entries_limit_per_query: 10000  #default is 5000
+   ```
+
+**Properties:**
+
+- Enforced by: Querier/Query Frontend
+- Retryable: No (query must be modified)
+- HTTP status: 400 Bad Request
+- Configurable per tenant: Yes
+
+### Error: Query would read too many bytes
+
+**Error message:**
+
+`the query would read too many bytes (query: <size>, limit: <limit>); consider adding more specific stream selectors or reduce the time range of the query`
+
+**Cause:**
+
+The estimated data volume for the query exceeds the configured limit. This is determined before query execution using index statistics.
+
+**Default configuration:**
+
+- `max_query_bytes_read`: 0B (disabled by default)
+
+**Resolution:**
+
+* **Add more specific stream selectors** to reduce data volume.
+* **Reduce the time range** of the query.
+
+* **Increase the limit** if resources allow:
+
+   ```yaml
+   limits_config:
+     max_query_bytes_read: 10GB
+   ```
+
+**Properties:**
+
+- Enforced by: Query Frontend
+- Retryable: No (query must be modified)
+- HTTP status: 400 Bad Request
+- Configurable per tenant: Yes
+
+### Error: Too many chunks (count)
+
+**Error message:**
+
+`the query hit the max number of chunks limit (limit: 2000000 chunks)`
+
+**Cause:**
+
+The number of chunks that the query would read exceeds the configured limit. This protects against queries that would scan excessive amounts of data and consume too much memory.
+
+**Default configuration:**
+
+- `max_chunks_per_query`: 2000000
+
+**Resolution:**
+
+* **Narrow stream selectors** to reduce the number of matching chunks:
+
+   ```logql
+   # Too broad
+   {job="app"}
+   
+   # More specific
+   {job="app", environment="production", namespace="api"}
+   ```
+
+* **Reduce the query time range** to scan fewer chunks.
+
+* **Increase the limit** if resources allow:
+
+   ```yaml
+   limits_config:
+     max_chunks_per_query: 5000000  #default is 2000000
+   ```
+
+**Properties:**
+
+- Enforced by: Query Frontend
+- Retryable: No (query must be modified)
+- HTTP status: 400 Bad Request
+- Configurable per tenant: Yes
+
+### Error: Stream matcher limits
+
+**Error message:**
+
+`max streams matchers per query exceeded, matchers-count > limit (1000 > 500)`
+
+**Cause:**
+
+The query contains too many stream matchers. This limit prevents queries with excessive complexity that could impact query performance.
+
+**Default configuration:**
+
+- `max_streams_matchers_per_query`: 1000
+
+**Resolution:**
+
+* **Simplify your query** by using fewer label matchers.
+* **Combine multiple queries** instead of using many OR conditions.
+* **Use regex matchers** to consolidate multiple values:
+
+   ```logql
+   # Instead of many matchers
+   {job="app1"} or {job="app2"} or {job="app3"}
```

Comment:
It is not valid to have multiple `{ ... }` in a query.

The error is for the amount of `name="value"` pairs inside the brackets.